### DISCUSSION TITLE
Rand_formula and group

### DISCUSSION THREAD

**Discussion by aston9090** [FrederickHuangLin/ANCOMBC#161] [Q&A]
Hello, 

Thank you for developing and maintaining the ANCOMBC package! 

I have been trying to analyse microbiome data. In the data, there are 5 groups, and each group contains 2 species, meaning that there are 10 different species in total. I would like to compare the abundance of microbes across groups while accounting for species (which explains more than 50% of variations).  I tried to run ancombc2 and got an error message. Please see below.

ANCOMBC2<-ancombc2(data=data, fix_formula = "Group", rand_formula = "1|Species", group= "Group" ,global = T, tax_level = "phylum")

Error in dimnames(x) <- dn : 
  length of 'dimnames' [2] not equal to array extent
In addition: There were 50 or more warnings (use warnings() to see the first 50)
Warning messages:
1: Model failed to converge with 8 negative eigenvalues: -9.1e-05 -1.0e-04 -1.0e-04 -1.0e-04 -1.0e-04 -3.4e-04 -3.5e-04 -3.5e-04
2: Model failed to converge with 2 negative eigenvalues: -8.1e-05 -8.1e-05
...

Any help and suggestions are greatly appreciated. Thank you!

**Comment by FrederickHuangLin** [comment]
Hi @aston9090,

I have reviewed the syntax you provided and have noticed the issue that may be causing problems.

It is important to specify the parenthesis for `random_formula`. For example, instead of `rand_formula = "1|Species"`, it should be written as `rand_formula = "(1|Species)"`.

Please let me know if you have any further questions or concerns.

Best regards,
Huang

**Reply by aston9090** [reply]
Hi Huang, 

Thank you for the reply! It solves the error now.  May I also ask if that's the right way to test whether there are significant differences across the groups, given that the groups contain different species?

Thanks again!

Best regards
Aston

**Reply by FrederickHuangLin** [reply]
I lack some context in your study. Based on the code you provided, you have an random intercept term for `species`, which means the same `species` has been measure repeatedly. If this is the case, then your code is correct. Otherwise, specifying `fix_formula = "Group + Species", rand_formula = NULL` might be more appropriate.
